src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/TelemetryInitializerBase.cs

```diff
@@ -33,15 +33,18 @@ public void Initialize(ITelemetry telemetry)
                 return;
             }
 
-            var request = context.Features.Get<RequestTelemetry>();
-
-            if (request == null)
+            lock (context)
             {
-                AspNetCoreEventSource.Instance.LogTelemetryInitializerBaseInitializeRequestNull();
-                return;
-            }
+                var request = context.Features.Get<RequestTelemetry>();
```

Comment:
is [this](https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/ead65975f0fa27650b40983265bad05fdea16a95/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/MvcDiagnosticsListener.cs#L24) call safe? Should it be locked as well? Any chance customer code running in parrellel with the MVC OnBegin?